blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpShardedMetricsHandler.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.outputs.handlers;
+
+import com.codahale.metrics.Timer;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.rackspacecloud.blueflood.exceptions.InvalidDataException;
+import com.rackspacecloud.blueflood.http.HttpRequestHandler;
+import com.rackspacecloud.blueflood.http.HttpResponder;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.io.IOContainer;
+import com.rackspacecloud.blueflood.io.LocatorIO;
+import com.rackspacecloud.blueflood.tracker.Tracker;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.utils.Metrics;
+import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.JsonNodeFactory;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.http.DefaultHttpResponse;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponse;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.jboss.netty.handler.codec.http.HttpVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+public class HttpShardedMetricsHandler implements HttpRequestHandler {
+    private static final Logger log = LoggerFactory.getLogger(HttpShardedMetricsHandler.class);
+    private final LocatorIO io = IOContainer.fromConfig().getLocatorIO();
+    private final Timer httpShardedMetricsFetchTimer = Metrics.timer(HttpShardedMetricsHandler.class,
+            "Handle HTTP request for metrics by shard");
+
+    @Override
+    public void handle(ChannelHandlerContext ctx, HttpRequest request) {
+        Tracker.getInstance().track(request);
+        final String tenantId = request.getHeader("tenantId");
+        final String shardId = request.getHeader("shard");
```

Comment:
@shintasmith @VinnyQ @ChandraAddala - if we accept this, would reading generic-sounding header names be an issue?  Should they be more specific?  Also, we'll need to see how the "tenantId" would match up with how we run repose and tenanted access.